source/aggregation-tutorials.txt

```diff
@@ -1,12 +1,104 @@
+
+.. meta::
+   :keywords: node.js, code example, runnable app
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
 .. toctree::
 
    /aggregation-tutorials/one-to-one-join.txt
 
-Read the following tutorials to learn how perform specific tasks using
-aggregation:
+Overview
+--------
+
+Aggregation tutorials provide detailed explanations of common
+aggregation tasks in a step-by-step format. The tutorials are adapted
+from examples in the `Practical MongoDB Aggregations book
+<https://www.practical-mongodb-aggregations.com/>`__ by Paul Done.
+
+Each tutorial includes the following sections:
+
+- **Introduction**, which describes the purpose and common use cases of the
+  aggregation type. This section also describes the example and desired
+  outcome that the tutorial demonstrates.
+
+- **Before You Get Started**, which describes the necessary databases,
+  collections, and sample data that you must insert before building the
+  aggregation pipeline and performing the aggregation.
+
+- **Tutorial**, which describes how to build and run the aggregation
+  pipeline. This section describes each stage of the completed
+  aggregation tutorial, and then explains how to run and interpret the
+  output of the aggregation.
+
+At the end of each aggregation tutorial, you can find a link to a fully
+runnable Node.js code file that you can run in your environment.
+
+.. tip::
+
+   To learn more about performing aggregations, see the
+   :ref:`node-aggregation` guide.
+
+.. _node-agg-tutorial-template-app:
+
+Aggregation Template App
+------------------------
+
+Before you begin following an aggregation tutorial, set up an app that
+you can use to insert sample data into MongoDB and run the aggregation
+pipeline.
+
+.. tip:: 
+   
+   To learn how to install the driver and connect to MongoDB,
+   see the :ref:`node-quick-start-download-and-install` and
+   :ref:`node-quick-start-create-deployment` steps of the
+   Quick Start guide.
+
+Once you install the driver, create a file called
+``agg_tutorial.js``. Paste the following code in this file to create an
+app template for the aggregation tutorials:
+
+.. literalinclude:: /includes/aggregation/template-app.js
+   :language: javascript
+   :copyable: true
+
+.. important::
+
+   In the preceding code, read the code comments to find the sections of
+   the code that you must modify for the tutorial you are following.
+
+For every tutorial, you must replace the connection string placeholder with
+your deployment's connection string. For example, if your connection string is
+``"mongodb+srv://mongodb-example:27017"``, your connection string assignment resembles
+the following:
+
+.. code-block:: javascript
+   :copyable: false
+
+   const uri = "mongodb+srv://mongodb-example:27017";
+
+To run the completed file after you modify the template for a
+tutorial, run the following command in your terminal:
+
+.. code-block:: bash
+
+   node agg_tutorial.js
```

Comment:
Note that running this script before making changes per an individual tutorial will generate a connection error.

Reply:
For reference:
<img width="987" alt="Screenshot 2023-11-17 at 4 11 22 PM" src="https://github.com/mongodb/docs-node/assets/148898879/90e5eb41-0cdd-46e8-a94f-c64faac40e20">